jetty-core/jetty-quic/jetty-quic-quiche/jetty-quic-quiche-foreign/src/main/java/org/eclipse/jetty/quic/quiche/foreign/ForeignQuicheConnection.java

```diff
@@ -909,15 +911,17 @@ public int drainClearBytesForStream(long streamId, ByteBuffer buffer) throws IOE
                     // If the ByteBuffer is direct, it can be used without any copy.
                     MemorySegment bufferSegment = MemorySegment.ofBuffer(buffer);
                     MemorySegment fin = scope.allocate(NativeHelper.C_CHAR);
-                    read = quiche_h.quiche_conn_stream_recv(quicheConn, streamId, bufferSegment, buffer.remaining(), fin);
+                    MemorySegment outErrorCode = scope.allocate(NativeHelper.C_LONG);
+                    read = quiche_h.quiche_conn_stream_recv(quicheConn, streamId, bufferSegment, buffer.remaining(), fin, outErrorCode);
```

Comment:
Done for the 2 variables, I prefer to keep the `quiche_conn_stream_recv` call in the `if (buffer.isDirect())` block otherwise we would have to re-check the buffer in `if (read > 0)`.